docs/services/general-info/deprecating-variables.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Envvar Deprecation"
+date: 2024-08-22T15:41:00+01:00
+weight: 20
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/services/general-info
+geekdocFilePath: deprecating-variables.md
+---
+
+{{< toc >}}
+
+## Deprecating Environment Variables
+
+Sometimes it is necessary to deprecate an environment variable to align the naming with conventions or remove it at all. We therefore added annotations to automate the *documentation* process.
+
+The relevant annotations in the envvar struct tag are:
+
+* `deprecationVersion`\
+  The release an envvar is announced for deprecation.
+* `removalVersion`\
+  The version it is finally going to be removed is defined via the mandatory placeholder `%%NEXT_PRODUCTION_VERSION%%`, not an actual version number.
+* `deprecationInfo`\
+  The reason why it got deprecated.
+* `deprecationReplacement`\
+  Only if it is going to be replaced, not necessary if removed.
+
+{{< hint warning >}}
+During the development cycle, the value for the `removalVersion` must be set to `%%NEXT_PRODUCTION_VERSION%%`.  This placeholder will be removed by the real version number during the production releasing process.
+{{< /hint >}}
+
+For the documentation only to show the correct value for the `removalVersion`, our docs helper scripts will automatically generate the correct version to be printed in the documentation. If `%%NEXT_PRODUCTION_VERSION%%` is found in the query, it will be replaced with `next-prod`, else the value found is used.
```

Comment:
```suggestion
For the documentation to show the correct value for the `removalVersion`, our docs helper scripts will automatically generate the correct version to be printed in the documentation. If `%%NEXT_PRODUCTION_VERSION%%` is found in the query, it will be replaced with `next-prod`, else the value found is used.
```